### DISCUSSION TITLE
I2C "Mask Writing" to a Register (using bitmasks)?

### DISCUSSION THREAD

**Discussion by AL-M0ND** [micropython/micropython#15402] [RP2040 / Pico]
Hello all,

I am using a Raspberry Pi Pico H (RP2040) to control a target MC over I2C. I'm trying to write to specific bits in the target's registers. Using I2C.writeto_mem() writes to the whole register/memory address.
I saw [this thread]( https://stackoverflow.com/questions/70682373/write-a-specific-bit-in-a-16bits-register-with-modbus-in-python) and am wondering if MicroPython has any similar functionality to a mask write over I2C using bitmasks?

Let me know if any clarification is needed.
Thanks

**Comment by peterhinch** [comment]
The normal approach is to keep a copy of the byte last written to the address. When a bit needs changing, modify that bit in the copy, then re-write it out via I2C.

The approach discussed in the link is to read the data back from the device, modify the bit and re-write it. This is rather inefficient and reading back may not be supported by the target hardware,
